tycho-integration-test/src/execution/execution_simulator.rs

```diff
@@ -199,68 +196,114 @@ impl ExecutionSimulator {
             timeout: None,
         };
 
-        let trace_options =
-            GethDebugTracingCallOptions { tracing_options, state_overrides, block_overrides: None };
-
-        // We use debug_traceCall because it provides more meaningful simulation error results than
-        // other forms of simulation. (eth_call, using the provider)
-        // It is also already being used in the testing SDK, and supports overrides and simulating
-        // on past blocks, if we would need to do that in the future.
-        let trace_result: Value = provider
-            .client()
-            .request("debug_traceCall", (tx, BlockId::from(block_number), trace_options))
-            .await?;
-
-        // Check for error in response, print the full trace only on failure
-        if let Some(error) = trace_result.get("error") {
-            error!("=== Transaction Trace (FAILURE) ===");
-            crate::execution::traces::print_call_trace(&trace_result, 0).await;
-            error!("=== End Trace ===");
-
-            // Try to find the deepest revert reason in the trace
-            let reason =
-                if let Some(deepest_reason) = Self::find_deepest_revert_reason(&trace_result) {
-                    // Try to decode it if it looks like hex data
-                    if deepest_reason.starts_with("0x") {
-                        self.decode_error(&deepest_reason)
-                            .await?
-                    } else {
-                        deepest_reason
-                    }
-                } else if let Some(data) = error
-                    .get("data")
-                    .and_then(|d| d.as_str())
-                {
-                    self.decode_error(data).await?
-                } else if let Some(message) = error
-                    .get("message")
-                    .and_then(|m| m.as_str())
-                {
-                    message.to_string()
-                } else {
-                    "UnknownError".to_string()
-                };
-
-            return Ok(SimulationResult::Revert { reason });
+        let (simulation_ids, simulation_inputs): (Vec<String>, Vec<SimulationInput>) =
+            inputs.into_iter().unzip();
+
+        let block_id = BlockId::from(block.number());
+
+        let client = ClientBuilder::default().http(self.rpc_url.parse()?);
+        let mut batch = client.new_batch();
```

Comment:
So this initialized an "empty batch" and then we need to add to it one-by-one the calls that we want - did I understand right? So in theory we could mix endpoints?